File: web/src/stores/task.ts
/**
 * 任务状态管理
 */
import { defineStore } from "pinia";
import { ref } from "vue";
import { taskApi } from "@/api/task";
import type { Task } from "@/types/task";
import { ElMessage } from "element-plus";

export const useTaskStore = defineStore("task", () => {
  const tasks = ref<Task[]>([]);
  const currentTask = ref<Task | null>(null);
  const loading = ref(false);

  /**
   * 加载任务列表
   */
  const fetchTasks = async () => {
    loading.value = true;
    try {
      const response = await taskApi.getTasks();
      if (response.data.success) {
        tasks.value = response.data.data.tasks;
      }
    } catch (error) {
      ElMessage.error("加载任务列表失败");
      console.error(error);
    } finally {
      loading.value = false;
    }
  };

  /**
   * 加载我的任务
   */
  const fetchMyTasks = async () => {
    loading.value = true;
    try {
      const response = await taskApi.getMyTasks();
      if (response.data.success) {
        tasks.value = response.data.data.tasks;
      }
    } catch (error) {
      ElMessage.error("加载我的任务失败");
      console.error(error);
    } finally {
      loading.value = false;
    }
  };

  /**
   * 加载任务详情
   */
  const fetchTask = async (taskId: number) => {
    loading.value = true;
    try {
      const response = await taskApi.getTask(taskId);
      if (response.data.success) {
        currentTask.value = response.data.data.task;
      }
    } catch (error) {
      ElMessage.error("加载任务详情失败");
      console.error(error);
    } finally {
      loading.value = false;
    }
  };

  /**
   * 创建任务
   */
  const createTask = async (taskData: any) => {
    loading.value = true;
    try {
      const response = await taskApi.createTask(taskData);
      if (response.data.success) {
        ElMessage.success("任务创建成功");
        await fetchTasks(); // 刷新列表
        return response.data.data.task;
      }
    } catch (error: any) {
      ElMessage.error(error.response?.data?.message || "创建任务失败");
      throw error;
    } finally {
      loading.value = false;
    }
  };

  /**
   * 更新任务
   */
  const updateTask = async (taskId: number, taskData: any) => {
    loading.value = true;
    try {
      const response = await taskApi.updateTask(taskId, taskData);
      if (response.data.success) {
        ElMessage.success("任务更新成功");
        await fetchTasks(); // 刷新列表
        return response.data.data.task;
      }
    } catch (error: any) {
      ElMessage.error(error.response?.data?.message || "更新任务失败");
      throw error;
    } finally {
      loading.value = false;
    }
  };

  /**
   * 删除任务
   */
  const deleteTask = async (taskId: number) => {
    loading.value = true;
    try {
      const response = await taskApi.deleteTask(taskId);
      if (response.data.success) {
        ElMessage.success("任务删除成功");
        await fetchTasks(); // 刷新列表
        return true;
      }
    } catch (error: any) {
      ElMessage.error(error.response?.data?.message || "删除任务失败");
      throw error;
    } finally {
      loading.value = false;
    }
  };

  return {
    tasks,
    currentTask,
    loading,
    fetchTasks,
    fetchMyTasks,
    fetchTask,
    createTask,
    updateTask,
    deleteTask,
  };
});

File: app/src/api/v1/task.py
"""任务管理 API 端点"""

from typing import Annotated, List
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.orm import Session
from loguru import logger

from src.middleware.auth import CurrentUser, check_user_permission
from src.dao.task_dao import TaskDAO
from src.types.standard_response import StandardResponse
from src.types.task_models import TaskCreate, TaskUpdate, TaskResponse
from ..utils import exception_wrapper, get_db_session

router = APIRouter(prefix="/tasks", tags=["任务管理"])


def get_task_dao(db_session: Annotated[Session, Depends(get_db_session)]):
    """获取任务 DAO 依赖"""
    return TaskDAO(db_session)


@router.post("/", response_model=StandardResponse, status_code=status.HTTP_201_CREATED)
@exception_wrapper(catch_http_exc=True)
async def create_task(
    task_data: TaskCreate,
    current_user: CurrentUser = Depends(check_user_permission()),
    task_dao: TaskDAO = Depends(get_task_dao),
):
    """创建任务"""
    task = task_dao.create_task(
        title=task_data.title,
        creator_id=current_user.user_id,
        description=task_data.description,
        status=task_data.status,
        priority=task_data.priority,
        due_date=task_data.due_date,
    )

    return StandardResponse(
        success=True, message="任务创建成功", data={"task": TaskResponse.from_orm(task).dict()}
    )


@router.get("/", response_model=StandardResponse)
@exception_wrapper(catch_http_exc=True)
async def get_tasks(
    skip: int = 0,
    limit: int = 100,
    current_user: CurrentUser = Depends(check_user_permission()),
    task_dao: TaskDAO = Depends(get_task_dao),
):
    """获取任务列表（分页）"""
    tasks = task_dao.get_all_tasks(skip=skip, limit=limit)

    return StandardResponse(
        success=True,
        message="获取任务列表成功",
        data={
            "tasks": [TaskResponse.from_orm(t).dict() for t in tasks],
            "total": len(tasks),
            "skip": skip,
            "limit": limit,
        },
    )


@router.get("/my", response_model=StandardResponse)
@exception_wrapper(catch_http_exc=True)
async def get_my_tasks(
    current_user: CurrentUser = Depends(check_user_permission()),
    task_dao: TaskDAO = Depends(get_task_dao),
):
    """获取我的任务"""
    tasks = task_dao.get_tasks_by_creator(current_user.user_id)

    return StandardResponse(
        success=True,
        message="获取我的任务成功",
        data={"tasks": [TaskResponse.from_orm(t).dict() for t in tasks]},
    )


@router.get("/{task_id}", response_model=StandardResponse)
@exception_wrapper(catch_http_exc=True)
async def get_task(
    task_id: int,
    current_user: CurrentUser = Depends(check_user_permission()),
    task_dao: TaskDAO = Depends(get_task_dao),
):
    """获取任务详情"""
    task = task_dao.get_task_by_id(task_id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")

    return StandardResponse(
        success=True, message="获取任务详情成功", data={"task": TaskResponse.from_orm(task).dict()}
    )


@router.put("/{task_id}", response_model=StandardResponse)
@exception_wrapper(catch_http_exc=True)
async def update_task(
    task_id: int,
    task_data: TaskUpdate,
    current_user: CurrentUser = Depends(check_user_permission()),
    task_dao: TaskDAO = Depends(get_task_dao),
):
    """更新任务"""
    task = task_dao.get_task_by_id(task_id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")

    # 检查权限：只有创建者可以更新
    if task.creator_id != current_user.user_id:
        raise HTTPException(status_code=403, detail="无权限修改此任务")

    update_data = task_data.dict(exclude_unset=True)
    success = task_dao.update_task(task_id, **update_data)

    if not success:
        raise HTTPException(status_code=500, detail="更新任务失败")

    updated_task = task_dao.get_task_by_id(task_id)
    return StandardResponse(
        success=True, message="任务更新成功", data={"task": TaskResponse.from_orm(updated_task).dict()}
    )


@router.delete("/{task_id}", response_model=StandardResponse)
@exception_wrapper(catch_http_exc=True)
async def delete_task(
    task_id: int,
    current_user: CurrentUser = Depends(check_user_permission()),
    task_dao: TaskDAO = Depends(get_task_dao),
):
    """删除任务"""
    task = task_dao.get_task_by_id(task_id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")

    # 检查权限：只有创建者或管理员可以删除
    if task.creator_id != current_user.user_id and current_user.role not in [
        "admin",
        "super_admin",
    ]:
        raise HTTPException(status_code=403, detail="无权限删除此任务")

    success = task_dao.delete_task(task_id)
    if not success:
        raise HTTPException(status_code=500, detail="删除任务失败")

    return StandardResponse(success=True, message="任务删除成功", data={})
